src/clients/dotnet/TigerBeetle.Tests/UInt128Tests.cs

```diff
@@ -4,149 +4,102 @@
-            var expected = new byte[16] { 86, 52, 18, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
-
-            Assert.IsTrue(expected.SequenceEqual(expected.ToUInt128().ToArray()));
-            Assert.IsTrue(expected.SequenceEqual(BigInteger.Parse("123456", NumberStyles.HexNumber).ToUInt128().ToArray()));
-            Assert.IsTrue(expected.SequenceEqual(new Guid(expected).ToUInt128().ToArray()));
-            Assert.IsTrue(expected.SequenceEqual(new UInt128(0, 0x123456).ToArray()));
-        }
-
-#if !NET7_0_OR_GREATER
-
-        [TestMethod]
-        public void HashCode()
-        {
-            UInt128 a = 100;
-            UInt128 b = 101;
-
-            Assert.AreNotEqual(a.GetHashCode(), b.GetHashCode());
-        }
-
-        [TestMethod]
-        public void Equals()
-        {
-            // Same type:
-            Assert.IsTrue(UInt128.Zero.Equals(UInt128.Zero));
-            Assert.IsTrue(((UInt128)1).Equals((UInt128)1));
-            Assert.IsTrue(new UInt128(0UL, ulong.MaxValue).Equals(new UInt128(0UL, ulong.MaxValue)));
-
-            // Implicit conversion:
-            Assert.IsTrue(UInt128.Zero.Equals(0));
-            Assert.IsTrue(((UInt128)1).Equals(1));
-            Assert.IsTrue(new UInt128(0UL, ulong.MaxValue).Equals(ulong.MaxValue));
-
-            // Explicit conversion:
-            Assert.IsTrue(UInt128.Zero.Equals((int)0));
-            Assert.IsTrue(((UInt128)1).Equals((int)1));
-            Assert.IsTrue(new UInt128(0UL, long.MaxValue).Equals(long.MaxValue));
-
-        }
-
-        [TestMethod]
-        public void Operators()
-        {
-            var a = new UInt128(100, 200);
-            var b = new UInt128(100, 200);
-
-            Assert.AreEqual(a, b);
-            Assert.IsTrue(a.Equals(b));
-            Assert.IsTrue(b.Equals(a));
-            Assert.IsTrue(a == b);
-            Assert.IsTrue(b == a);
-            Assert.IsFalse(a != b);
-            Assert.IsTrue(a != UInt128.Zero);
-        }
-#endif
+        byte[] array = new byte[16] { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x10 };
+        UInt128 value = array.ToUInt128();
+
+        Assert.IsTrue(value.ToArray().SequenceEqual(array));
+        Assert.IsTrue(array.SequenceEqual(value.ToArray()));
+        Assert.IsTrue(value.Equals(array.ToUInt128()));
+    }
+
+    [TestMethod]
+    [ExpectedException(typeof(ArgumentNullException))]
+    public void NullArrayConvertion()
+    {
+        byte[] array = null!;
+        _ = array.ToUInt128();
+    }
+
+    [TestMethod]
+    [ExpectedException(typeof(ArgumentException))]
+    public void EmptyArrayConvertion()
+    {
+        byte[] array = new byte[0];
+        _ = array.ToUInt128();
+    }
+
+    [TestMethod]
+    [ExpectedException(typeof(ArgumentException))]
+    public void InvalidArrayConvertion()
+    {
+        byte[] array = new byte[17];
+        _ = array.ToUInt128();
+    }
+
+
+    [TestMethod]
+    public void BigIntegerConvertion()
```

Comment:
In a separate PR, let's also add tests for small numbers, like `0`, and also a test for a _negative_ BigInteger (which we expect to throw)